### PR TITLE
Posts and projects

### PR DESCRIPTION
**Pull request by triszt4n**
### What's new
* Post author can add a featured image onto post that will be present on the post and on preview
* Modified README for further instructions for authors
* Projects page (also with image)

### CODE REVIEW

**Comment by triszt4n** [comment]
* This branch builds onto `feat/improve-header`, so bear it in mind
* I've readied it for review, but there are dummy files in the branch remaining
* I'm open for any changes

**Comment by balintkiraly** [comment]
May we change the target branch to `feat/improve-header`?! Just to make the review easier

**Comment by OmTheTurtle** [comment]
Took a quick look at the preview on netlify, my initial thoughts:

Blogflow:
- the images are too big, and generally not so important so they should be smaller (especially since a lot of posts don't have a lead)
- I'm thinking about a 2-column layout for desktop, with a bit smaller font size and thinner font weight for the title
- the horizontal gap between the posts should be larger (especially on mobile they aren't separate enough)
- I really like the dot matrix design on the pictures, I'd experiment with putting them only to the right and the bottom of the image

Projects:
- I'd experiment with a 2-column layout, since here the image has more significance
- some data about the project could fit into the card (esp. if we use a wider card) like the main tech (something like we have on the GitHub organization page).

**Comment by triszt4n** [comment]
Applied all your suggestions @OmTheTurtle, but forgot to apply all the new fields of a project to the project show page. (Also, gonna add a website field so that one could add not only the github repo but the actual website of the project)

**Comment by triszt4n** [comment]
I guess I could've used something else than a `.ts` file for constants like the pek_url, because this change needed a whole new build, maybe environment variables should help it not to build again. (Gonna look up netlify envvars)

**Comment by OmTheTurtle** [comment]
> I guess I could've used something else than a `.ts` file for constants like the pek_url, because this change needed a whole new build, maybe environment variables should help it not to build again. (Gonna look up netlify envvars)

Yeah, env vars are probably the best solution for that :+1:

File: src/components/blog-components/BlogPreview.tsx [hunk 1]
```diff
@@ -1,52 +1,86 @@
...
-      <Box width={{ base: '100%', sm: '85%' }} zIndex="2" marginLeft={{ base: '0', sm: '5%' }} marginTop="5%">
-        <Link to={post.fields.slug}>
-          <Image
-            borderRadius="lg"
-            src={post.frontmatter.previewImgSrc ?? '../../blog-noimg.png'}
-            fallbackSrc="../../blog-noimg.png"
-            alt="Blog preview"
-            objectFit="contain"
+const BlogPreview: React.FC<BlogPreviewProps> = ({ post, isBig }) => {
+  const result = getImage(post.frontmatter.featuredImage)
```
Comment (balintkiraly) [inline src/components/blog-components/BlogPreview.tsx:29]: Can we rename this variable to image or featuredImage or something like that

File: src/components/indexpage-components/PekPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      github: string
+      featuredImage: ImageDataLike
+      status: string
+      techs: string
+    }
+  }
+}
+
+const PekPreview: React.FC<PekProps> = ({ project }) => {
+  const result = getImage(project.frontmatter.featuredImage)
```
Comment (balintkiraly) [inline src/components/indexpage-components/PekPreview.tsx:26]: Same here

File: src/components/project-components/ProjectPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      return <CheckIcon color={color + tone} />
+    case 'Áll':
+      return <WarningIcon color={color + tone} />
+    default:
+      return <CircleIcon color={color + tone} />
+  }
+}
+
+const ProjectPreview: React.FC<ProjectForPreview> = ({ project }) => {
+  const result = getImage(project.frontmatter.featuredImage)
```
Comment (balintkiraly) [inline src/components/project-components/ProjectPreview.tsx:44]: and here

File: src/content/projects/xblog-next5.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+layout: project
+title: 'Kir-Dev Blog'
+lead: 'A Kir-Dev következő generációs blogja.'
+github: https://github.com/kir-dev/blog-next
+website: https://kir-dev.sch.bme.hu
+status: 'Áll red'
```
Comment (balintkiraly) [inline src/content/projects/xblog-next5.md:7]: I guess Gatsby can handle not just strings. Maybe we can use an object. It's ok to merge this and create an issue for it
```
status: {
  label: 'Áll',
  color: 'red',
}
```

File: src/templates/post.tsx [hunk 5]
```diff
@@ -18,24 +23,83 @@ interface PostTemplateProps {
...
-          {data.markdownRemark.frontmatter.title}
-        </Heading>
-        {/* eslint-disable-next-line react/no-danger */}
-        <div dangerouslySetInnerHTML={{ __html: data.markdownRemark.html }} />
-      </Container>
-    </Post>
-  </IndexLayout>
-)
+const PostTemplate: React.FC<PostTemplateProps> = ({ data }) => {
+  const result = getImage(data.markdownRemark.frontmatter.featuredImage)
```
Comment (balintkiraly) [inline src/templates/post.tsx:40]: change to image

File: src/templates/post.tsx [hunk 6]
```diff
@@ -18,24 +23,83 @@ interface PostTemplateProps {
...
+  // const pic =
```
Comment (balintkiraly) [inline src/templates/post.tsx:41]: Delete this comment please

**Review by balintkiraly — APPROVED** [review]

**Review by OmTheTurtle — APPROVED** [review]

**Pull request merged**